src/content/1.5/code/agda/full.agda

```diff
@@ -0,0 +1,195 @@
+𝑝 : C → A
+𝑝 = {!!}
+𝑞 : C → B
+𝑞 = {!!}
+
+{- All `C`s that fit this pattern will be considered candidates for the product.
+There may be lots of them. For instance, let's pick, as our constituents, two
+Haskell types, `Int` and `Bool`, and get a sampling of candidates for their
+product. Here's one: `Int`. Can `Int` be considered a candidate for the product of
+`Int` and `Bool`? Yes, it can---and here are its projections:                    -}
+p₁ : Int → Int
+p₁ x = x
+q₁ : Int → Bool
+q₁ _ = true
+{- Here's another one: Int × Int × Bool. It's a tuple of three ele-
+ments, or a triple.
+-}
+
+{- Actually, this is an unfortunate example since it requires that we already know
+how to construct products. Let's pretend we don't and just take the following
+definition for granted for now.                                                  -}
+_×_ : ∀ (A : Set a) (B : Set b) → Set (a ⊔ b)
+A × B = Σ[ x ∈ A ] B
+infixr 2 _×_
+
+{- Here are two morphisms that make this example a legitimate
+candidate for the product `Int × Bool`:                                          -}
+p₂ : Int × Int × Bool → Int
+p₂ (x , _ , _) = x           -- We're using pattern matching on triples here...
+q₂ : Int × Int × Bool → Bool
+q₂ (_ , _ , b) = b           -- ...and here.
+
+{- But we want the "best" candidate `C`---i.e., the `C` such that for all other
+such candidates, say, `C'`, there is a morphism `m : C' → C` such that p' = p ∘ m
+and q' = q ∘ m. See the figure below.
+
+Another way of looking at these equations is that m factorizes p' and q'. To build
+some intuition, let's see that the pair `Int × Bool` with the two canonical
+projections, `fst` and `snd` is indeed better than the two candidates presented
+before. The mapping `m` for the first candidate is:                              -}
+m : Int → Int × Bool
+m x = (x , true)
+{- Indeed, the two projections can be reconstructed as:                          -}
+p₁' : Int → Int
+p₁' x = fst (m x)
+q₁' : Int → Bool
+q₁' x = snd (m x)
+{- The m for the second example, `C' = Int × Int × Bool` is similarly uniquely determined:
+-}
+m' : Int × Int × Bool → Int × Bool
+m' x = (p₂ x , q₂ x)
+{- Again, the two projections can be reconstructed as:
+-}
+p₂' : Int × Int × Bool → Int
+p₂' x = fst (m' x)
+q₂' : Int × Int × Bool → Bool
+q₂' x = snd (m' x)
+
+factorizer : (C → A) → (C → B) → C → A × B
+factorizer p q = λ x → (p x , q x)
+
+
+{- The product A × B:
+
+  A         B
+  |\       /|
+  |  𝑝   𝑞  |
+  |   \ /   |
+  𝑝'   C    𝑞'   (C = A × B = greatest lower bound of {A, B})
+   \   |   /
+    \  𝑚  /
+     \ | /
+       C'
+
+The diagram illustrates the universal property that defines the product C.
+It is the object that has projection maps 𝑝 : C → A and 𝑞 : C → B and is such
+that for every other object C', with projection maps 𝑝' : C' → A, 𝑞' : C' → B
+there exists a morphism 𝑚 that factorizes 𝑝' and 𝑞' as follows: 𝑝' = 𝑝 ∘ 𝑚
+and 𝑞' = 𝑞 ∘ 𝑚.                                                                -}
```

Comment:
Not an review remark - just a note to myself: I would love to see definition of universal property of product (and coproduct) expressed using Sigma and Pi :)

Reply:
I could easily add that, if you think it could be useful.

Reply:
added